profile-widget/profilescene.cpp

```diff
@@ -138,40 +138,60 @@ void ProfileScene::clear()
 	eventItems.clear();
 }
 
-static bool ppGraphsEnabled()
+static bool ppGraphsEnabled(const struct divecomputer *dc, bool simplified)
 {
-	return prefs.pp_graphs.po2 || prefs.pp_graphs.pn2 || prefs.pp_graphs.phe;
+	return simplified ? (dc->divemode == CCR && prefs.pp_graphs.po2)
+			  : (prefs.pp_graphs.po2 || prefs.pp_graphs.pn2 || prefs.pp_graphs.phe);
 }
 
 // Update visibility of non-interactive chart features according to preferences
-void ProfileScene::updateVisibility(bool diveHasHeartBeat)
+void ProfileScene::updateVisibility(bool diveHasHeartBeat, bool simplified)
 {
-#ifndef SUBSURFACE_MOBILE
-	pn2GasItem->setVisible(prefs.pp_graphs.pn2);
-	po2GasItem->setVisible(prefs.pp_graphs.po2);
-	pheGasItem->setVisible(prefs.pp_graphs.phe);
-
-	const struct divecomputer *currentdc = d ? get_dive_dc_const(d, dc) : nullptr;
-	bool setpointflag = currentdc && currentdc->divemode == CCR && prefs.pp_graphs.po2;
-	bool sensorflag = setpointflag && prefs.show_ccr_sensors;
-	o2SetpointGasItem->setVisible(setpointflag && prefs.show_ccr_setpoint);
-	ccrsensor1GasItem->setVisible(sensorflag);
-	ccrsensor2GasItem->setVisible(currentdc && sensorflag && currentdc->no_o2sensors > 1);
-	ccrsensor3GasItem->setVisible(currentdc && sensorflag && currentdc->no_o2sensors > 2);
-	ocpo2GasItem->setVisible(currentdc && currentdc->divemode == PSCR && prefs.show_scr_ocpo2);
-
-	heartBeatItem->setVisible(prefs.hrgraph && diveHasHeartBeat);
-#endif
-	diveCeiling->setVisible(prefs.calcceiling);
-	decoModelParameters->setVisible(prefs.decoinfo);
-#ifndef SUBSURFACE_MOBILE
-	for (DiveCalculatedTissue *tissue: allTissues)
-		tissue->setVisible(prefs.calcalltissues && prefs.calcceiling);
-	percentageItem->setVisible(prefs.percentagegraph);
-#endif
-	meanDepthItem->setVisible(prefs.show_average_depth);
-	reportedCeiling->setVisible(prefs.dcceiling);
-	tankItem->setVisible(prefs.tankbar);
+	const struct divecomputer *currentdc = get_dive_dc_const(d, dc);
+	if (!currentdc)
+		return;
+	bool ppGraphs = ppGraphsEnabled(currentdc, simplified);
+
+	if (simplified) {
```

Comment:
I really like the idea of the 'simplified' profile - once again, a 12" iPad may want to show the whole thing, and this design makes this so much easier.